### PR TITLE
[Feature] #16, #17 : Add ToolDictionaryMainView UI

### LINKED ISSUE #16: [Feature] Construct ToolDictionaryMainView's Searchbar
# 유저는 장비를 검색할 수 있다
- 장비를 검색하는 검색창 UI를 구성한다

## 🤔 Completion conditions

- [ ] Construct ToolDictionaryMainView's Searchbar UI like the design of a sketch

### LINKED ISSUE #17: [Feature] Construct ToolDictionaryMainView's GridView
# 유저는 장비군을 확인할 수 있다
- 장비군을 확인할 수 있는 Grid UI를 구성한다

## 🤔 Completion conditions

- [ ] Construct ToolDictionaryMainView's GridView UI like the design of a sketch ( low-fi )

## 👀 Related issues

- Related issue1 ( #16 )

### PR DESCRIPTION
**Pull request by rriver2**
# Issue Number
🔒 Close #16 
🔒 Close #17 

## Changes

- Changed files
    - campair/campair.xcodeproj/project.pbxproj
    - campair/campair/View/ToolDictionary/ToolDictionaryMainView.swift

## New features

- Create the ToolDictionaryMainView's SearchBar UI 
- Create the ToolDictionaryMainView's GridView UI 

## Checklist

- [x] Is the branch you are merging on correct?
- [x] Do you comply with coding conventions?
- [x] Are there any changes not related to PR?
- [x] Has my code been self-reviewed?

### CODE REVIEW

File: campair/campair/View/ToolDictionary/ToolDictionaryMainView.swift [hunk 1]
```diff
@@ -0,0 +1,92 @@
+//
+//  ToolDictionaryMainView.swift
+//  campair
+//
+//  Created by 이가은 on 2022/06/12.
+//
+
+import SwiftUI
+
+let columns: [GridItem] = [
```
Comment (sustainable-git) [inline campair/campair/View/ToolDictionary/ToolDictionaryMainView.swift:10]: You should not declare global property here...
Use nested style instead

File: campair/campair/View/ToolDictionary/ToolDictionaryMainView.swift [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+                        HStack(spacing : 0){
+                            Image(systemName: "magnifyingglass")
+                                .padding(.horizontal, 10)
+                            TextField("검색어를 입력하세요.", text: $searchWord)
+                        }
+                    }
+                }
+                Text("어서와요!(가안)\n장비를 알아보아요")
+                    .bold()
+                    .font(.system(size:25))
```
Comment (sustainable-git) [inline campair/campair/View/ToolDictionary/ToolDictionaryMainView.swift:40]: Font size should not be fixed.
Please consider accessibility.

File: campair/campair/View/ToolDictionary/ToolDictionaryMainView.swift [hunk 3]
```diff
@@ -0,0 +1,92 @@
...
+                        .padding(.trailing, 20)
+                    ZStack(alignment: .leading){
+                        Rectangle()
+                            .stroke()
+                            .background(Color.white)
+                            .frame(height: 34)
+                        HStack(spacing : 0){
+                            Image(systemName: "magnifyingglass")
+                                .padding(.horizontal, 10)
+                            TextField("검색어를 입력하세요.", text: $searchWord)
```
Comment (sustainable-git) [inline campair/campair/View/ToolDictionary/ToolDictionaryMainView.swift:34]: Our coding convention has a rule about using explicit self
Please use `self.` in front of `$searchWord`

**Review by sustainable-git — CHANGES_REQUESTED** [review]
Please check review :)

File: campair/campair/View/ToolDictionary/ToolDictionaryMainView.swift (hunk 3 above)
Comment (rriver2) [reply campair/campair/View/ToolDictionary/ToolDictionaryMainView.swift:34]: thank you. I fixed it

File: campair/campair/View/ToolDictionary/ToolDictionaryMainView.swift (hunk 1 above)
Comment (rriver2) [reply campair/campair/View/ToolDictionary/ToolDictionaryMainView.swift:10]: thank you.  I fixed it

File: campair/campair/View/ToolDictionary/ToolDictionaryMainView.swift (hunk 2 above)
Comment (rriver2) [reply campair/campair/View/ToolDictionary/ToolDictionaryMainView.swift:40]: Thank you. But I want to ask berry's opinion. After her hi-fi drawing, I will reflect this.  @vveryberry

File: campair/campair/View/ToolDictionary/ToolDictionaryMainView.swift [hunk 4]
```diff
@@ -0,0 +1,92 @@
...
+//  ToolDictionaryMainView.swift
+//  campair
+//
+//  Created by 이가은 on 2022/06/12.
+//
+
+import SwiftUI
+
+struct ToolDictionaryMainView: View {
+    @State private var searchWord : String = ""
```
Comment (unnnyong) [inline campair/campair/View/ToolDictionary/ToolDictionaryMainView.swift:11]:
```suggestion
    @State private var searchWord = ""
```

Int, Double, Float 가 아닌경우에는 Swift의 타입 속도을 믿고 생략해도 좋을 것 같아요 ~

File: campair/campair/View/ToolDictionary/ToolDictionaryMainView.swift [hunk 5]
```diff
@@ -0,0 +1,92 @@
...
+                        HStack(spacing : 0){
+                            Image(systemName: "magnifyingglass")
+                                .padding(.horizontal, 10)
+                            TextField("검색어를 입력하세요.", text: self.$searchWord)
+                        }
+                    }
+                }
+                Text("어서와요!(가안)\n장비를 알아보아요")
+                    .bold()
+                    .font(.system(size:25))
```
Comment (unnnyong) [inline campair/campair/View/ToolDictionary/ToolDictionaryMainView.swift:40]:
```suggestion
                    .font(.system(size: 25))
```

보통의 Swift 코드 스타일로는 Colon 뒤에 스페이스 하나가 추천됩니다 ~

File: campair/campair/View/ToolDictionary/ToolDictionaryMainView.swift [hunk 6]
```diff
@@ -0,0 +1,92 @@
...
+                    
+                    ForEach(0..<16){ index in
+                        ToolBoxView(image: Image(systemName: "hammer"), name: "캠핑페어")
+                    }
+                }
+                Spacer()
+            }
+            .padding(.horizontal, 25)
+            .padding(.top, 30)
+            .navigationTitle("")
```
Comment (unnnyong) [inline campair/campair/View/ToolDictionary/ToolDictionaryMainView.swift:54]: Title이 나중에 추가되나요 ? 아니면 이유가 있어서 `""`가 들어간것인가용 ?

File: campair/campair/View/ToolDictionary/ToolDictionaryMainView.swift [hunk 7]
```diff
@@ -0,0 +1,92 @@
...
+            .navigationBarHidden(true)
+        }
+    }
+}
+
+struct ToolBoxView: View {
+    let image : Image
+    let name : String
```
Comment (unnnyong) [inline campair/campair/View/ToolDictionary/ToolDictionaryMainView.swift:62]: 이곳에서만 사용된다면 private 설정이 좋을 것 같아요 ~
```suggestion
    private let image : Image
    private let name : String
```

File: campair/campair/View/ToolDictionary/ToolDictionaryMainView.swift [hunk 8]
```diff
@@ -0,0 +1,92 @@
...
+    
+    var body: some View {
+        
+        Button {
+
```
Comment (unnnyong) [inline campair/campair/View/ToolDictionary/ToolDictionaryMainView.swift:67]: 추후에 액션이 추가된다면 TODO로 잊지않을 수 있어요 ~ !
```suggestion
            // TODO: Add Handling
```

File: campair/campair/View/ToolDictionary/ToolDictionaryMainView.swift [hunk 9]
```diff
@@ -0,0 +1,92 @@
...
+                    Text("Logo")
+                        .padding(.trailing, 20)
+                    ZStack(alignment: .leading){
+                        Rectangle()
+                            .stroke()
+                            .background(Color.white)
+                            .frame(height: 34)
+                        HStack(spacing : 0){
+                            Image(systemName: "magnifyingglass")
+                                .padding(.horizontal, 10)
```
Comment (unnnyong) [inline campair/campair/View/ToolDictionary/ToolDictionaryMainView.swift:33]: 사용되는 아이콘? SFSymbol들을 어디에서 관리하면 어때요 ?

```swift
enum Symbol: String {
    case search = "magnifyingglass"
}
```

**Comment by unnnyong** [comment]
diff 양도, PR 본문도 심플해서 리뷰하기 너무 편했어요 ~~~~~~~ ! 감사합니다 ❤️

File: campair/campair/View/ToolDictionary/ToolDictionaryMainView.swift (hunk 5 above)
Comment (rriver2) [reply campair/campair/View/ToolDictionary/ToolDictionaryMainView.swift:40]: I fixed it thank you !

File: campair/campair/View/ToolDictionary/ToolDictionaryMainView.swift (hunk 6 above)
Comment (rriver2) [reply campair/campair/View/ToolDictionary/ToolDictionaryMainView.swift:54]: Oh, I did't delete this code. Thank you for finding this. I fixed it ~ !

File: campair/campair/View/ToolDictionary/ToolDictionaryMainView.swift (hunk 7 above)
Comment (rriver2) [reply campair/campair/View/ToolDictionary/ToolDictionaryMainView.swift:62]: I think this part will be modified to move on to the next view after the entity is decided.

File: campair/campair/View/ToolDictionary/ToolDictionaryMainView.swift (hunk 8 above)
Comment (rriver2) [reply campair/campair/View/ToolDictionary/ToolDictionaryMainView.swift:67]: Good ~~! Thank you for letting me know!

File: campair/campair/View/ToolDictionary/ToolDictionaryMainView.swift (hunk 4 above)
Comment (rriver2) [reply campair/campair/View/ToolDictionary/ToolDictionaryMainView.swift:11]: > PR 본문

Thank you young !

Comment (rriver2) [reply campair/campair/View/ToolDictionary/ToolDictionaryMainView.swift:11]: didn't know this! Thank you for explan this ~ Thank you yong ><

**Review by unnnyong — APPROVED** [review]
🚀 🚀 🚀 🚀

**Pull request merged**